src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -470,20 +470,20 @@ bool TryDeleteRange(out DateTime? next)
                         return end < to;
                     }
                 }
+            }
+        }
 
-                DateTime? TryGetNextBaseline()
-                {
-                    var offset = slicer.TimeSeriesKeySlice.Size - sizeof(long);
-                    *(long*)(slicer.TimeSeriesKeySlice.Content.Ptr + offset) = Bits.SwapBytes(baseline.Ticks / 10_000);
-
-                    foreach (var (_, tvh) in table.SeekByPrimaryKeyPrefix(slicer.TimeSeriesPrefixSlice, slicer.TimeSeriesKeySlice, 0))
-                    {
-                        return GetBaseline(tvh.Reader);
-                    }
+        public DateTime? BaselineOfNextSegment(Slice key, Slice prefix, Table table, DateTime baseline)
+        {
+            var offset = key.Size - sizeof(long);
+            *(long*)(key.Content.Ptr + offset) = Bits.SwapBytes(baseline.Ticks / 10_000);
```

Comment:
Hm... here you are mutating the key for the slice, no?
Wouldn't that cause problems? For that matter, isn't that _already_ the right value here?

Reply:
I agree it looks wired, but this is how the existed code behave. 

let's look into that after this PR is merged, since I'm not sure it is a bug and we need this PR to continue the testing.